test/libsolidity/InlineAssembly.cpp

```diff
@@ -291,27 +320,27 @@ BOOST_AUTO_TEST_CASE(oversize_string_literals)
 
 BOOST_AUTO_TEST_CASE(assignment_after_tag)
 {
-	BOOST_CHECK(successParse("{ let x := 1 { tag: =: x } }"));
+	BOOST_CHECK(successParse("{ let x := 1 { 7 tag: =: x } }"));
```

Comment:
I want clear-cut semantics in order to deal with dirty code.

So, when you jump into `tag:`, you need to make sure that the topmost stack elements are

```
---------
| something
--------
| x
--------
|
| other local variables
|
---------------
``` ?

And then, after the assignment before `} }`, the conceptual stack elements would be
```
--------
| x := something
--------
|
| other local variables
|
---------------
``` ?

In that case, I have a question.  Is it always guaranteed that all local variables are together at the bottom of the conceptual stack?  Or, can I put local variables on top of the `deposit()`s?

Is `{ 2 { let x := 1} pop}` valid?

How does this affect the local variables declared around this?